CMakeLists.txt

```
@@ -51,5 +54,26 @@ target_compile_options(MolSim
 )
 
 list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/modules)
+#Option to build Documentation. Can be disabled by calling cmake -D BUILD_DOC=OFF ..
+option(BUILD_DOC "Build documentation" ON)
+if(BUILD_DOC)
+include(doxygen)
+
+
+#Check if Doxygen is installed
+find_package(Doxygen)
+if (DOXYGEN_FOUND)
+    file(MAKE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/docs)
+    # create doxygen target
+    add_custom_target(doc_doxygen
+            COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile
+            COMMENT "Building HTML Documentation")
+
+
 
-include(doxygen)
\ No newline at end of file
+else ()
+    message("-- Doxygen needs to be installed to generate the doxygen documentation.")
```

I made the same mistake when I took this course 5 years ago during my Bachelor's 😅🥲. So you get the same feedback as I did:

_Please check the CMake documentation_ for the [message](https://cmake.org/cmake/help/latest/command/message.html) function.

You want to change this (and other occurrences) to:
```suggestion
    message(STATUS "Doxygen needs to be installed to generate the doxygen documentation.")
```